app/src/main/java/io/github/trojan_gfw/igniter/servers/presenter/ServerListPresenter.java

```diff
@@ -206,6 +212,30 @@ public void gotoScanQRCode() {
         mView.gotoScanQRCode();
     }
 
+    @Override
+    public void pingAllProxyServer(List<TrojanConfig> configList) {
+        for (TrojanConfig config : configList) {
+            mDataManager.pingTrojanConfigServer(config, new ServerListDataSource.PingCallback() {
```

Comment:
Better to wrap the call with a `WeakReference` of `ServerListPresenter`. If user exits `ServerListActivity` while testing ping, this will lead to an memory leak. Or you could interrupt the test before exiting.

Reply:
Ok.

Reply:
Well, this way of wrapping does not prevent the memory leak😂 It is still a strong reference.
Maybe you could refer to `ServerListPresenter.SubscribeCallback`.

Reply:
> Well, this way of wrapping does not prevent the memory leak It is still a strong reference.
> Maybe you could refer to `ServerListPresenter.SubscribeCallback`.

yes. I got it.

Reply:
Hi @TchaikovDriver 
If still a strong reference to ServerListPresenter.
I have a question, Could you help me? 
scene:

If the mDataManager work thread has a long time IO work on the background.
This time, click the back key to destroy the ServerListActivity.
the mDataManager call back to UI thread ServerListActivity.
No crash. no error logs.

Could help to explain it?